client/src/details/SFSClientImpl.cpp

```diff
@@ -391,7 +422,8 @@ std::string SFSClientImpl<ConnectionManagerT>::GetBaseUrl() const
     {
         return *m_customBaseUrl;
     }
-    return "https://" + m_accountId + "." + std::string(c_apiDomain);
+
+    return "https://" + SFSUrlComponents::UrlEscape(m_accountId + "." + std::string(c_apiDomain), m_reportingHandler);

```

Comment:
I have never seen the host part being escaped/percent-encoded. Does libcurl specifically say that it handles escaped/encoded hostnames? If it doesn't, then the encoded/escaped hostname you provide will be used for DNS lookup and will most likely fail or worse, will resolve to an IP that we don't want (think malicious users squatting on such odd domains).

Perhaps you should just validate the url using libcurl and throw a specific error if that fails. See https://curl.se/libcurl/c/libcurl-url.html.

Also, for encoding/escaping the path/query components, check with SFS service team that they or their http stack (.net?) expects it.

Reply:
That class looks really useful, thanks.
Using it does seem like a better alternative.

I don't believe SFS expects any kind of encoding. Service strings are just alphanumeric characters, and that's why I was at first just validating and avoiding sending them over. But Jeff mentions we should have the service validate instead, and we can only do that through encoding, otherwise the whole URL breaks and we can't even get the service to validate the strings.

Reply:
I've repurposed this PR to only validate request strings are not empty, and will implement a url builder with this curl class in a different PR.